hazelcast-jclouds/src/test/java/com/hazelcast/jclouds/ComputeServiceBuilderTest.java

```diff
@@ -79,4 +90,217 @@ public void test_destroy_calls_computeService_destroy() throws Exception {
         builder.destroy();
         verify(mockComputeServiceContext).close();
     }
+
+
+    @Test
+    public void test_buildZonesConfig_parses_multiple_zones() throws Exception {
+        Map<String, Comparable> properties = new HashMap<String, Comparable>();
+        properties.put("zones", "zone1,zone2,zone3");
+        ComputeServiceBuilder builder = new ComputeServiceBuilder(properties);
+        builder.buildRegionZonesConfig();
+        assertEquals(3, builder.getZonesSet().size());
+        assertTrue(builder.getZonesSet().contains("zone1"));
+        assertTrue(builder.getZonesSet().contains("zone2"));
+        assertTrue(builder.getZonesSet().contains("zone3"));
+        assertFalse(builder.getZonesSet().contains("zone4"));
+    }
+
+    @Test
+    public void test_buildZonesConfig_parse_single_zone() throws Exception {
+        Map<String, Comparable> properties = new HashMap<String, Comparable>();
+        properties.put("zones", "zone1");
+        ComputeServiceBuilder builder = new ComputeServiceBuilder(properties);
+        builder.buildRegionZonesConfig();
+        assertEquals(1, builder.getZonesSet().size());
+        assertTrue(builder.getZonesSet().contains("zone1"));
+        assertFalse(builder.getZonesSet().contains("zone2"));
+    }
+
+    @Test
+    public void test_buildZonesConfig_parse_multiple_regions() throws Exception {
+        Map<String, Comparable> properties = new HashMap<String, Comparable>();
+        properties.put("regions", "region1,region2,region3");
+        ComputeServiceBuilder builder = new ComputeServiceBuilder(properties);
+        builder.buildRegionZonesConfig();
+        assertEquals(3, builder.getRegionsSet().size());
+        assertTrue(builder.getRegionsSet().contains("region1"));
+        assertTrue(builder.getRegionsSet().contains("region2"));
+        assertTrue(builder.getRegionsSet().contains("region3"));
+        assertFalse(builder.getRegionsSet().contains("region4"));
+    }
+
+    @Test
+    public void test_buildZonesConfig_parse_single_region() throws Exception {
+        Map<String, Comparable> properties = new HashMap<String, Comparable>();
+        properties.put("regions", "region1");
+        ComputeServiceBuilder builder = new ComputeServiceBuilder(properties);
+        builder.buildRegionZonesConfig();
+        assertEquals(1, builder.getRegionsSet().size());
+        assertTrue(builder.getRegionsSet().contains("region1"));
+        assertFalse(builder.getRegionsSet().contains("region2"));
+    }
+
+    @Test
+    public void test_isNodeInsideZones_when_zone_set_is_empty() throws Exception {
+        ComputeServiceBuilder builder = new ComputeServiceBuilder(new HashMap<String, Comparable>());
+        builder.buildRegionZonesConfig();
+
+        assertTrue(builder.isNodeInsideZones(
+                getRunningNodeMetaDataAtLocation(LocationScope.ZONE, "testZone")));
```

Comment:
`testZone ` is used to differ verified zone from other zone names. I named zones which are not verified as `zone1`,`zone2` etc.